packages/fuels-core/src/types/transaction_builders.rs

```diff
@@ -906,4 +941,110 @@ mod tests {
             status: MessageStatus::Unspent,
         }
     }
+
+    fn given_inputs(num_inputs: u8) -> Vec<Input> {
+        (0..num_inputs)
+            .map(|i| {
+                let bytes = [i; 32];
+                let coin = CoinType::Coin(Coin {
+                    utxo_id: UtxoId::new(bytes.into(), 0),
+                    owner: Bech32Address::new("fuel", bytes),
+                    ..Default::default()
+                });
+                Input::resource_signed(coin)
+            })
+            .collect()
+    }
+
+    fn given_witnesses(num_witnesses: usize) -> Vec<Witness> {
+        repeat_with(Witness::default).take(num_witnesses).collect()
+    }
+
+    #[derive(Default)]
+    struct MockDryRunner {
+        c_param: ConsensusParameters,
+    }
+
+    #[cfg_attr(not(target_arch = "wasm32"), async_trait)]
+    impl DryRunner for MockDryRunner {
+        async fn dry_run_and_get_used_gas(&self, _: FuelTransaction, _: f32) -> Result<u64> {
+            Ok(0)
+        }
+        fn consensus_parameters(&self) -> &ConsensusParameters {
+            &self.c_param
+        }
+        async fn min_gas_price(&self) -> Result<u64> {
+            Ok(0)
+        }
+    }
+
+    #[tokio::test]
+    async fn create_tb_builder_witness_indexes_set_correctly() -> Result<()> {
```

Comment:
Shouldn't it be tx builder instead of tb builder?
```suggestion
    async fn create_tx_builder_witness_indexes_set_correctly() -> Result<()> {
```